bundler/lib/bundler/cli/platform.rb

```diff
@@ -9,8 +9,8 @@ def initialize(options)
 
     def run
       platforms, ruby_version = Bundler.ui.silence do
-        locked_ruby_version = Bundler.locked_gems && Bundler.locked_gems.ruby_version&.gsub(/p\d+\Z/, "")
```

Comment:
Mmmm this is the kind of change I wanted to avoid with `MaxChainLength: 1`. Should we revert it?

Reply:
How about splitting?
```ruby
        locked_ruby_version = Bundler.locked_gems&.ruby_version
        locked_ruby_version = locked_ruby_version&.gsub(/p\d+\Z/, "")
```

Reply:
If think this is the simplest

```ruby
ruby_version = if Bundler.locked_gems
  Bundler.locked_gems.ruby_version&.gsub(/p\d+\Z/, "")
else
  Bundler.definition.ruby_version&.single_version_string
end
```

And then below replace

```ruby
locked_ruby_version || gemfile_ruby_version
```

with

```ruby
ruby_version
```

Reply:
Then `ruby_version` will be shared with the outer scope.

Is `ruby_version` ok by just assigned?
```ruby
      ruby_version = nil
      platforms = Bundler.ui.silence do
        ruby_version = if Bundler.locked_gems
          Bundler.locked_gems.ruby_version&.gsub(/p\d+\Z/, "")
        else
          Bundler.definition.ruby_version&.single_version_string
        end
        Bundler.definition.platforms.map {|p| "* #{p}" }
      end
```

Reply:
BTW, I can't find a way to align `else` and `end` with the LHS of the assignment, in ruby-mode.el.

Reply:
It seems fine! Actually, the `Bundler.ui.silence` block was introduced a long time ago at https://github.com/rubygems/bundler/commit/33ed52e703b3fa9337787344f0425063b33a2f91, and I believe we probably don't need it anymore, so we can assign both variables directly?

Reply:
I wasn't sure if it has any effects now.
And `platforms` seems unused in `if options[:ruby]`.